code/lib/codemod/src/transforms/__tests__/mdx-to-csf.test.ts

```diff
@@ -0,0 +1,412 @@
+
+    # hello
+
+    export const args = { bla: 1 };
+
+    <Meta of={FoobarStories} />
+
+    world {2 + 1}
+
+    <Story of={FoobarStories.Foo} />
+
+    <Story of={FoobarStories.Unchecked} />
+
+  `);
+
+  const [csfFileName, csf] = fs.writeFileSync.mock.calls[0];
+  expect(csfFileName).toMatchInlineSnapshot(`Foobar.stories.js`);
+  expect(csf).toMatchInlineSnapshot(`
+      const args = { bla: 1 };
+
+      export default {
+        title: 'foobar',
+      };
+
+      export const Foo = {
+        render: () => 'bar',
+        name: 'foo',
+      };
+
+      export const Unchecked = {
+        render: Template.bind({}),
+        name: 'Unchecked',
+
+        args: {
+          ...args,
+          label: 'Unchecked',
+        },
+      };
+
+    `);
+});
+
+test('extract all meta parameters', () => {
+  const input = dedent`
+      import { Meta } from '@storybook/addon-docs';
+
+      export const args = { bla: 1 };
+      
+      <Meta title="foobar" args={{...args}} parameters={{a: '1'}} />
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      const args = { bla: 1 };
+
+      export default {
+        title: 'foobar',
+
+        args: {
+          ...args,
+        },
+
+        parameters: {
+          a: '1',
+        },
+      };
+
+    `);
+});
+
+test('extract all story attributes', () => {
+  const input = dedent`
+      import { Meta, Story } from '@storybook/addon-docs';
+
+      export const args = { bla: 1 };
+      
+      <Meta title="foobar" />
+
+      <Story name="foo">bar</Story>
+      
+      <Story 
+        name="Unchecked"
+        args={{
+          ...args,
+          label: 'Unchecked',
+        }}>    
+        {Template.bind({})}
+      </Story>
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      const args = { bla: 1 };
+
+      export default {
+        title: 'foobar',
+      };
+
+      export const Foo = {
+        render: () => 'bar',
+        name: 'foo',
+      };
+
+      export const Unchecked = {
+        render: Template.bind({}),
+        name: 'Unchecked',
+
+        args: {
+          ...args,
+          label: 'Unchecked',
+        },
+      };
+
+    `);
+});
+
+test('story child is jsx', () => {
+  const input = dedent`
+      import { Canvas, Meta, Story } from '@storybook/addon-docs';
+      import { Button } from './button';
+      
+      <Story name="Primary">
+        <Button>
+          <div>Hello!</div>
+        </Button>
+      </Story>
+    `;
+
+  jscodeshift({ source: input, path: 'Foobar.stories.mdx' });
+
+  const [, csf] = fs.writeFileSync.mock.calls[0];
+
+  expect(csf).toMatchInlineSnapshot(`
+      import { Button } from './button';
+      export default {};
+
+      export const Primary = {
+        render: () => (
+          <Button>
```

Comment:
Should this file `import React from 'react'`?

Reply:
From React 17 it is not needed anymore, and also back-ported to the latest 16 release. 
There are ESLint rules to automatically fix this if you need it, so I kind of feel better with leaving it out of this codemod.